superset_patchup/views.py

```diff
@@ -0,0 +1,61 @@
+"""
+Additional extensions to SuperSet views
+"""
+
+import json
+from flask_appbuilder import expose
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+import superset
+
+from . import (VERSION, __version__)
+
+class SupersetKetchupApiView(superset.views.base.BaseSupersetView):
+
+    """
+    Extensions to Superset API
+    """
+
+    def __init__(self):
+        self.route_base = "/superset-ketchup/api"
+
+    @superset.views.base.api
+    @expose('/version/', methods=['GET'])
+    def version(self):
+        """Returns *superset-api* version"""
+        return self.json_response({ 'version': VERSION, 'versionStr': __version__ })
+
+    @superset.views.base.api
+    @expose('/all_dashboards/', methods=['GET'])
+    def all_dashboards(self):
+        """
+        Extension to allow API access to information *about* dashboards.
+        See: /fave_dashboards/, etc.
+        Returns the ids, links, names, and URLs of the dashboards a user has access to.
+        """
+
+        dashboards = superset.db.session.query(
+            superset.models.core.Dashboard)
+
+        # Query only dashboards where we have at least some access - note that this exact query
+        # filter is used in the core Superset view to prune visible dashboards.
+        # Logic is basically "dashboard owned by user or at least one slice accessible by user"
+        # See: https://github.com/apache/incubator-superset/blob/0.27/superset/views/core.py#L154
+        dashboards = (superset.views.core.DashboardFilter('slice',
+        	SQLAInterface(superset.models.core.Dashboard)).apply(dashboards, lambda: None))
+
+        payload = []
+        for dashboard in dashboards:
+
+            value = {
+                'id': dashboard.id,
+                'dashboard_link': dashboard.dashboard_link(),
+                'dashboard_title': dashboard.dashboard_title,
+                'url': dashboard.url,
+            }
+
+            payload.append(value)
+
+        return self.json_response(payload)
+
+# Hook up the view
+superset.appbuilder.add_view_no_menu(SupersetKetchupApiView)
```

Comment:
Want to double-check with ansible stuffs whether or not this loaded automatically when the superset-patchup module gets loaded or if we need to shove it into __init__.py

Reply:
See later patch - figured it out.  Superset loads all its views last, the rightest thing for us I think is to use the FLASK_APP_MUTATOR hook superset gives us for all our mutating needs.